pkg/controller/certificate-shim/sync.go

```diff
@@ -322,6 +322,10 @@ func buildCertificates(
 		}
 	case *gwapi.Gateway:
 		for i, l := range ingLike.Spec.Listeners {
+			if (l.Protocol != gwapi.HTTPSProtocolType) {
```

Comment:
Wouldn't you still need this if the Protocol was TLS?

Reply:
@fortra-cloudops-platform, no I don't think so if my understanding of the following snippet is correct, as the Gateway API provides a `TLSProtocolType` const

https://github.com/kubernetes-sigs/gateway-api/blob/746db5eaeac75d3fed8ddb9d60d56cfb4b150702/apis/v1beta1/gateway_types.go#L314-L332

Reply:
I think the point is you also shouldn't `continue` if `l.Protocol` is `TLSProtocolType` which sounds plausible.

https://gateway-api.sigs.k8s.io/v1alpha2/guides/tls/#listeners-and-tls

Reply:
Thanks for the feedback!

I was going by the test cases, where I only saw `HTTPSProtocolType` being used - but I think you are right, and `TLSProtocolType` should also be supported and work already. I've adapted the code accordingly, and also added a test case to explicitly validate it works also for TLS. This new test:

- passes in the current `master`
- fails with my original change set
- passes with the new change set